engine/proofview_monad.ml

```diff
@@ -149,13 +149,25 @@ module Info = struct
     CList.map_append (collapse_tree n) f
 end
 
+module StateStore = Store.Make(struct end)
+
+(* let (set_state, get_state) = StateDyn.Easy.make_dyn "goal_state" *)
+
+type goal = Evar.t
+type goal_with_state = Evar.t * StateStore.t
```

Comment:
Please use `Dyn.Map` here instead of `Store.t`.

Reply:
Please fix `Store`.